bot/exts/info/patreon.py

```diff
@@ -0,0 +1,129 @@
+    "We use the money we get to offer excellent prizes for all of our events. Prizes like t-shirts, "
+    "stickers, microcontrollers that support CircuitPython, or maybe even a mechanical keyboard.\n\n"
+    "You can read more about how Patreon donations help us, and consider donating yourself, on our patreon page "
+    "[here](https://pydis.com/patreon)!"
+)
+
+# List of tuples containing tier number and Discord role ID.
+# Ordered from highest tier to lowest.
+PATREON_TIERS: list[tuple[int, int]] = [
+    (3, Roles.patreon_tier_3),
+    (2, Roles.patreon_tier_2),
+    (1, Roles.patreon_tier_1),
+]
+
+
+def get_patreon_tier(member: discord.Member) -> int:
+    """
+    Get the patreon tier of `member`.
+
+    A patreon tier of 0 indicates the user is not a patreon.
+    """
+    for tier, role_id in PATREON_TIERS:
+        if has_role_id(member, role_id):
+            return tier
+    return 0
+
+
+class Patreon(commands.Cog):
+    """Cog that shows patreon supporters."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+        self.current_monthly_supporters.start()
+
+    @commands.Cog.listener()
+    async def on_member_update(self, before: discord.Member, after: discord.Member) -> None:
+        """Send a message when someone receives a patreon role."""
+        old_patreon_tier = get_patreon_tier(before)
+        new_patreon_tier = get_patreon_tier(after)
+
+        if not new_patreon_tier > old_patreon_tier:
+            return
+
+        message = (
+            f":tada: {after.mention} just became a **tier {new_patreon_tier}** patron!\n"
+            "Support us on Patreon: https://pydis.com/patreon"
+        )
+        channel = await get_or_fetch_channel(Channels.meta)
+        await channel.send(message)
+
+    async def send_current_supporters(self, channel: discord.abc.Messageable, automatic: bool = False) -> None:
+        """Send the current list of patreon supporters, sorted by tier level."""
+        guild = self.bot.get_guild(Guild.id)
+
+        embed_list = []
+        for tier, role_id in PATREON_TIERS:
+            role = guild.get_role(role_id)
+
+            # Filter out any members where this is not their highest tier.
+            patrons = [member for member in role.members if get_patreon_tier(member) == tier]
+            patron_names = [f"• {patron.name}#{patron.discriminator}" for patron in patrons]
+
+            embed = discord.Embed(
+                title=role.name,
+                description="\n".join(patron_names),
+                colour=role.colour
+            )
+            embed_list.append(embed)
+
+        main_embed = discord.Embed(
+            title="Patreon Supporters - Monthly Update" if automatic else "Patreon Supporters",
+            description=(
+                PATREON_INFORMATION +
+                "\n\nThank you to the users listed below who are already supporting us!"
+            ),
+        )
+
+        await channel.send(embeds=(main_embed, *embed_list))
```

Comment:
According to [the docs](https://discordpy.readthedocs.io/en/latest/api.html#discord.abc.Messageable.send), the embeds kwarg should be a list and not a tuple.

Reply:
I opened https://github.com/Rapptz/discord.py/issues/8460 as this shouldn't matter and it's type hinted as allowing sequences anyway